src/pages/SimpleViewer.jsx

```diff
@@ -1,62 +1,99 @@
-import React, { useState } from 'react';
+import React, { useState, useContext } from 'react';
 
 import { Row, Col } from 'react-bootstrap';
 import Dropzone from 'react-dropzone';
 import { FaCloudUploadAlt } from 'react-icons/fa';
-import Snackbar from '@material-ui/core/Snackbar';
+import Button from '@material-ui/core/Button';
 
 import API from '@/API';
+import { useVisibility } from '@/utils/cache';
+import AlertContext from '@/context/alert';
+import config from '@/config.json';
+import parseMessage from '@/utils/parseMessage';
+import usePageStatus from '@/utils/usePageStatus';
+import useMessageStore from '@/stores/useMessageStore';
 
-// import AppConfig from '../AppConfig';
-import Loading from '../components/loading/Loading';
-import AnswersetView from '../components/shared/answersetView/AnswersetView';
-// import AnswersetStore from './stores/messageAnswersetStore';
-import useMessageStore from '../stores/useMessageStore';
-import config from '../config.json';
-import parseMessage from '../utils/parseMessage';
+import AnswersetView from '@/components/shared/answersetView/AnswersetView';
 
 export default function SimpleViewer(props) {
   const { user } = props;
-  // We only read the communications config on creation
-  // const [appConfig, setAppConfig] = useState(new AppConfig(config));
   const [messageSaved, setMessageSaved] = useState(false);
-  const [loading, toggleLoading] = useState(false);
-  // const [user, setUser] = useState({});
-  // const [concepts, setConcepts] = useState([]);
-  const [errorMessage, setErrorMessage] = useState('');
-  const [loadingMessage, setLoadingMessage] = useState('');
-  const [showSnackbar, toggleSnackbar] = useState(false);
   const messageStore = useMessageStore();
+  const pageStatus = usePageStatus(false);
+  const visibility = useVisibility();
+  const displayAlert = useContext(AlertContext);
+
+  async function askQuestion() {
```

Comment:
I think that all of the error handling in askQuestion and uploadMessage should display alerts instead of setting the pageStatus to failed. The whole page is still functional, so we should just let the user know something went wrong and let them continue using the page.